src/appleseed/foundation/utility/poison.h

```diff
@@ -122,6 +135,9 @@ class PoisonImpl<double>
     {
         x = FP<double>::snan();
     }
+    static void do_always_poison(double& x){
+        x = FP<double>::snan();
```

Comment:
What's the difference with do_poison ?

Reply:
There is none, technically one could also wrap a poison call here.